File: 1328-break-a-palindrome/1328-break-a-palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
        if(palindrome.length() == 1) return "";
        
        int n = palindrome.length();
        bool flag = false;
        
        for(int i = 0; i < n; ++i) {
            if(palindrome[i] != 'a' && i != n / 2) {
                palindrome[i] = 'a';
                flag = true;
                break;
            }
        }
        if(flag == false) {
            palindrome[n - 1] = 'b';
        }
        return palindrome;
    }
};

File: 1328-break-a-palindrome/1328-break-a-palindrome.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        n = len(palindrome)
        
        if n == 1: return ""
        
        flag = False
        toList = list(palindrome)
        
        for i in range(n):
            if palindrome[i] != 'a' and i != n // 2:
                toList[i] = 'a'
                flag = True
                break
        
        if flag == False:
            toList[n - 1] = 'b'
        
        return ''.join(toList)